rasa/shared/core/generator.py

```diff
@@ -52,12 +52,13 @@ class TrackerWithCachedStates(DialogueStateTracker):
     def __init__(
         self,
         sender_id: Text,
-        slots: Optional[List[Slot]],
+        slots: Optional[Iterable[Slot]],
```

Comment:
The constructor in `DialogueStateTracker seems to accept only `Optional[Iterable[Slot]]` [here.](https://github.com/RasaHQ/rasa/blob/f104371249ac64476a50196d174bb2e637b6a2a5/rasa/shared/core/trackers.py#L184)

Reply:
ohhh I guess I was confused because I was looking at `DialogueStateTracker.from_dict` which accepts `Optional[List[Slot]]`. Can we change it there as well for consistency?

Reply:
Done: https://github.com/RasaHQ/rasa/pull/10965/commits/5eca012da282219bf072befb916ab04cb441ae55